File: App/Models/Entities/Bill.cs
using App.Models.Entities.Identities;
using MaxV.Base;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace App.Models.Entities
{
    public class Bill : BaseEntity<int>
    {
        public override void SetValueUpdate(string updateAt)
        {
            BillDetails.ToList().ForEach(e => e.SetValueUpdate(updateAt));
            base.SetValueUpdate(updateAt);
        }
        public virtual Customer Customer { get; set; }
        public int CustomerId { get; set; }
        public virtual User UserPayment { get; set; }
        public string UserPaymentId { get; set; }
        public double? TotalPrice { get; set; }
        public double? DiscountPrice { get; set; }
        public double? PaymentAmount { get; set; }
        public virtual ICollection<BillDetail> BillDetails { get; set; }
    }
}

File: admin/src/app/models/view-models/bill.ts
import { BaseViewModel ,BillDetail} from "@app/models";

export class Bill  extends BaseViewModel<number> {
    customerPhoneNumber?: string = null;
    customerFullName?: string = null;
    customerAddress?: string = null;
    userPaymentId?: string = null;
    userPaymentUserName?: string = null;
    totalPrice: number = null;
    discountPrice: number = null;
    paymentAmount?: number = null;
    billDetails: BillDetail[] = [];
}
